src/appleseed/foundation/math/filter.h

```diff
@@ -43,6 +43,108 @@
 namespace foundation
 {
 
+//
+// Base class for 1D reconstruction filters.
+//
+// The filters are not normalized (they don't integrate to 1 over their domain).
+// The return value of evaluate() is undefined if (x) is outside the filter's domain.
+//
+
+template <typename T>
```

Comment:
Do we really need to template this ? If we now we are only going to use floats, can't we simply switch to float ?

Reply:
Out of consistency with the 2D filters, I suppose.

The code in appleseed.foundation aims at being generic, that's why a lot of it is templated.